### PR TITLE
Added search bar in policy files

### PR DESCRIPTION
**Pull request by chaitali-mane**
Signed-off-by: Chaitali Mane [cmane@progress.com](cmane@progress.com)

### :nut_and_bolt: Description: What code changed, and why?
We need search functionality in policy files list

### :chains: Related Resources
https://github.com/chef/automate/issues/5240

### :+1: Definition of Done
Added search bar functionality for policy files

### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
Steps for Policyfiles tab:
Go To Infrastructure -> Chef Servers -> click server name-> click on org name -> Policyfiles.

For cypress Build
https://github.com/chef/automate/tree/master/e2e

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
1. Search bar
search-policy

2. No data after search
search-policy1

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html [hunk 1]
```diff
@@ -2,13 +2,18 @@
   <chef-loading-spinner *ngIf="policyFilesListLoading" size="50"></chef-loading-spinner>
   <app-empty-state *ngIf="authFailure" moduleTitle="policyfiles" (resetKeyRedirection)="resetKeyTabRedirection($event)"></app-empty-state>
   <ng-container *ngIf="!policyFilesListLoading && !authFailure">
-    <div data-cy="empty-list" class="empty-section" *ngIf="!policyFiles.length">
+    <div class="search-create-container">
+      <app-infra-search-bar (searchButtonClick)="searchPolicyFiles($event)" placeHolder="Policy">
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:6]: It should be ``Policy files```. Please confirm with the UX team once

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.ts [hunk 2]
```diff
@@ -51,7 +55,7 @@ export class PolicyFilesComponent implements OnInit, OnDestroy {
     .subscribe(([ getPolicyFilesSt, allPolicyFilesState]) => {
       if (getPolicyFilesSt === EntityStatus.loadingSuccess && !isNil(allPolicyFilesState)) {
         this.policyFiles = allPolicyFilesState;
-        this.policyFilesListLoading = false;
+                this.policyFilesListLoading = false;
```
Comment (himanshi-chhabra) [inline components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.ts:58]: update the alignment

File: e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts [hunk 3]
```diff
@@ -130,5 +131,29 @@ describe('infra policy file', () => {
...
+          cy.get('[data-cy=policy-file-table-container]').then(body => {
+            if (body.text().includes(policyFileName)) {
+              cy.get('[data-cy=policy-file-table-container]')
+              .contains(policyFileName).should('exist');
+            }
+          });
+
+          cy.get('[data-cy=search-filter]').clear();
+          cy.get('[data-cy=search-entity]').click();
+          // getPolicyFile().then((response) => {
```
Comment (himanshi-chhabra) [inline e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts:150]: remove comments

File: components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.ts (hunk 2 above)
Comment (chaitali-mane) [reply components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.ts]: Incorporated review comment.

File: e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts [hunk 4]
```diff
@@ -130,5 +131,26 @@ describe('infra policy file', () => {
...
+        if (checkResponse(response)) {
+          cy.get('[data-cy=search-filter]').type(policyFileName);
+          cy.get('[data-cy=search-entity]').click();
+          cy.get('[data-cy=policy-file-table-container]').then(body => {
+            if (body.text().includes(policyFileName)) {
+              cy.get('[data-cy=policy-file-table-container]')
+              .contains(policyFileName).should('exist');
+            }
+          });
+
```
Comment (Venkatesh-rengasamy) [inline e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts:147]: remove new line.

**Review by himanshi-chhabra — APPROVED** [review]

File: e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts (hunk 4 above)
Comment (chaitali-mane) [reply e2e/cypress/integration/ui/infra-proxy/infra-policy-file-list.spec.ts:147]: we keeping new line as separator

**Review by Venkatesh-rengasamy — APPROVED** [review]

**Pull request merged**